src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/ArchiveRoutes.scala

```diff
@@ -0,0 +1,117 @@
+package ch.epfl.bluebrain.nexus.kg.routes
+
+import akka.http.scaladsl.model.StatusCodes.{Created, OK}
+import akka.http.scaladsl.model.headers.Accept
+import akka.http.scaladsl.model.{HttpEntity, MediaTypes}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import ch.epfl.bluebrain.nexus.admin.client.types.Project
+import ch.epfl.bluebrain.nexus.iam.client.types._
+import ch.epfl.bluebrain.nexus.kg.KgError.{InvalidOutputFormat, UnacceptedResponseContentType}
+import ch.epfl.bluebrain.nexus.kg.archives.Archive._
+import ch.epfl.bluebrain.nexus.kg.config.AppConfig
+import ch.epfl.bluebrain.nexus.kg.directives.AuthDirectives.hasPermission
+import ch.epfl.bluebrain.nexus.kg.directives.PathDirectives._
+import ch.epfl.bluebrain.nexus.kg.directives.ProjectDirectives._
+import ch.epfl.bluebrain.nexus.kg.directives.QueryDirectives.outputFormat
+import ch.epfl.bluebrain.nexus.kg.marshallers.instances._
+import ch.epfl.bluebrain.nexus.kg.resources._
+import ch.epfl.bluebrain.nexus.kg.resources.syntax._
+import ch.epfl.bluebrain.nexus.kg.routes.OutputFormat.Tar
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import io.circe.Json
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+
+class ArchiveRoutes private[routes] (archives: Archives[Task])(
+    implicit acls: AccessControlLists,
+    project: Project,
+    caller: Caller,
+    config: AppConfig
+) {
+
+  private val responseType = MediaTypes.`application/x-tar`
+
+  /**
+    * Routes for archives. Those routes should get triggered after the following segment have been consumed:
+    * {prefix}/archives/{org}/{project}. E.g.: v1/archives/myorg/myproject
+    */
+  def routes: Route =
+    concat(
+      // Create an archive when id is not provided in the Uri (POST)
+      (post & pathEndOrSingleSlash) {
+        operationName(s"/${config.http.prefix}/archives/{}/{}") {
+          (hasPermission(write) & projectNotDeprecated) {
+            entity(as[Json]) { source =>
+              complete(archives.create(source).value.runWithStatus(Created))
```

Comment:
Missing `Location` header with 303 see other.

Reply:
That's on purpose. I wanted this API to be as similar as the rest of the API. As when you create a file in the system, you can fetch it back using the same endpoint as the PUT but with a GET, here you can use the `self` to fetch either the resource or it's binary content, depending on content negotiation HTTP headers.